syntax = "proto3";

package AgentTool;

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

service AgentToolController {
    rpc List(AgentToolListRequest) returns (stream AgentTool) {}
    rpc Create(AgentTool) returns (AgentTool) {}
    rpc Retrieve(AgentToolRetrieveRequest) returns (AgentTool) {}
    rpc Update(AgentTool) returns (AgentTool) {}
    rpc Destroy(AgentTool) returns (google.protobuf.Empty) {}
}

message AgentTool {
    string id = 1;
    string tool_name = 2;
    string args_schema = 3;
    string description = 4;
    google.protobuf.Timestamp created_at = 5;
    google.protobuf.Timestamp updated_at = 6;
}

message AgentToolListRequest {
}

message AgentToolRetrieveRequest {
    string id = 1;
}
